### DISCUSSION TITLE
Is it possible to run bigcode models like starcoder?

### DISCUSSION THREAD

**Discussion by wangjiyang** [ggml-org/llama.cpp#2130] [Ideas]
StarCoder show great performance on code generation and other coding stuff. However, bigcode models is very slow even with starcoder.cpp. Is it possible to run bigcode models like starcoder?

**Comment by KerfuffleV2** [comment]
Not with `llama.cpp` (at the moment). However, the GGML library which it uses as a backend has examples for various model types including StarCoder: https://github.com/ggerganov/ggml/tree/master/examples

The examples there generally don't have all the fancy features of `llama.cpp` (like advanced samplers) but they do allow at least running the model. I'm not sure what `starcoder.cpp` is based on. It might now be meaningfully different from the GGML StarCoder example. If it's already based on GGML then you probably aren't going to find anything much faster running on CPU anyway.
